DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/RevitExternalApplication.cs

```diff
@@ -40,27 +41,40 @@ public RevitExternalApplication()
     };
   }
 
+  /// <summary>
+  /// WARNING : We can't use any types from dlls that may have conflicts directly in this method
+  /// or this method will fail to load and the conflict detecting mechanism won't work.
+  ///
+  /// <para>
+  /// If you need to use types from a potentially conflicting dll, then use a different method
+  /// and make sure that it is called after the <see cref="DllConflictManager.DetectConflicts"/> call
+  /// </para>
+  /// </summary>
+  /// <param name="application"></param>
+  /// <returns></returns>
   public Result OnStartup(UIControlledApplication application)
   {
+    DllConflictManager conflictManager = new(new DllConflictManagmentOptionsLoader());
     try
     {
-      // POC: not sure what this is doing...  could be messing up our Aliasing????
       AppDomain.CurrentDomain.AssemblyResolve += OnAssemblyResolve;
-      _container = new AutofacContainer(new StorageInfo());
-      _container.PreBuildEvent += _container_PreBuildEvent;
-
-      // init DI
-      _container
-        .LoadAutofacModules(_revitSettings.ModuleFolders)
-        .AddSingletonInstance<RevitSettings>(_revitSettings) // apply revit settings into DI
-        .AddSingletonInstance<UIControlledApplication>(application) // inject UIControlledApplication application
-        .Build();
-
-      // resolve root object
-      _revitPlugin = _container.Resolve<IRevitPlugin>();
-      _revitPlugin.Initialise();
+      conflictManager.DetectConflicts();
+
+      InitializePlugin(application, conflictManager);
+
+      conflictManager.WarnUserOfPossibleConflicts();
```

Comment:
What happens if InitializePlugin() fails. Could InitializePlugin() not explode before being able to warn the user of the possible conflicts?

Reply:
I think the warning should be added in a `finally` block to ensure that whatever happens we tell the user if there were conflicts.

Reply:
We don't need to do this. Per the acceptance criteria, the plugin is warning users about potential conflicts in the "WarnUserOfPossibleConflicts" method. If intializePlugin fails due to a typeLoadException or MissingMethodException, then the exception will get handled in the catch block below by the conflictManger. 

The conflict manager will be able to tell the user that a conflict definitely did happen, as opposed to just warning that a conflict may happen.